### PR TITLE
refactor(tooling): Split Theme into `Theme` and `ThemePreference`

### LINKED ISSUE #4212: Split Theme into `Theme` and `ThemePreference`
We should split the Theme enum into two, so we can differenciate them when doing logic in components

### PR DESCRIPTION
**Pull request by marc2332**
Closes https://github.com/iotaledger/iota/issues/4212

### CODE REVIEW

**Review by evavirseda — CHANGES_REQUESTED** [review]
everything works smoothly! I just noticed that the local storage keys are formatted differently in some apps, would it be possible to standardize them in this PR? :pray:  the wallet and dashboard keys are different from the explorer, which includes 'iota-' in the key
image

**Review by evavirseda — APPROVED** [review]
<details>
    <summary>
        LGTM 
    </summary>
    
</details>

File: apps/core/src/components/providers/ThemeProvider.tsx [hunk 1]
```diff
@@ -12,40 +12,73 @@ interface ThemeProviderProps {
...
+
+        // In case of SystemPreference, listen for system theme changes
+        if (themePreference === ThemePreference.System) {
+            const handleSystemThemeChange = () => {
+                const systemTheme = getSystemTheme();
+                setSystemTheme(systemTheme);
+            };
+            const systemThemeMatcher = window.matchMedia('(prefers-color-scheme: dark)');
+            systemThemeMatcher.addEventListener('change', handleSystemThemeChange);
+            return () => systemThemeMatcher.removeEventListener('change', handleSystemThemeChange);
```
Comment (VmMad) [inline apps/core/src/components/providers/ThemeProvider.tsx:55]: This is not a watcher, is it? it doesn't update the system theme as i change it in my config

Comment (marc2332) [reply apps/core/src/components/providers/ThemeProvider.tsx:55]: Fixed 🙏🏻

**Review by VmMad — APPROVED** [review]
lgtm

**Pull request merged**